matchzoo/layers/matching_layer.py

```diff
@@ -0,0 +1,131 @@
+"""An implementation of Matching Layer."""
+import typing
+
+from keras import backend as K
+from keras.engine import Layer
+
+
+class MatchingLayer(Layer):
+    """Layer that computes a matching matrix between samples in two tensors."""
+
+    def __init__(self, normalize: bool = False,
+                 matching_type: str = 'dot', **kwargs):
+        """
+        :class:`MatchingLayer` constructor.
+
+        :param normalize: Whether to L2-normalize samples along the
+            dot product axis before taking the dot product.
+            If set to True, then the output of the dot product
+            is the cosine proximity between the two samples.
+        :param matching_type: the similarity function for matching
+        :param **kwargs: Standard layer keyword arguments.
+        """
+        super(MatchingLayer, self).__init__(**kwargs)
+        self.normalize = normalize
+        self.matching_type = matching_type
+        self.supports_masking = True
+        self.valid_matching_type = ['dot', 'mul', 'plus', 'minus', 'concat']
+        if matching_type not in self.valid_matching_type:
+            raise ValueError("{} is not a valid matching type,"
+                             " {} expected.".format(self.matching_type,
+                                                    self.valid_matching_type))
+
+    def build(self, input_shape: list):
+        """
+        Build the layer.
+
+        :param input_shape: the shapes of the input tensors,
+            for MatchingLayer we need tow input tensors.
+        """
+        # Used purely for shape validation.
+        if not isinstance(input_shape, list) or len(input_shape) != 2:
+            raise ValueError('A `MatchingLayer` layer should be called '
+                             'on a list of 2 inputs.')
```

Comment:
'a list of 2 inputs' .& 'input type should be a list'

Reply:
'a list of 2 inputs' contains 'input type of list'